rts/Sim/MoveTypes/Systems/GroundMoveSystem.cpp

```diff
@@ -66,33 +139,50 @@ void GroundMoveSystem::Update() {
     }
 	{
         SCOPED_TIMER("Sim::Unit::MoveType::4::ProcessCollisionEvents");
-        auto view = Sim::registry.view<GroundMoveType>();
-        view.each([](GroundMoveType& unitId){
-            CUnit* unit = unitHandler.GetUnit(unitId.value);
-            AMoveType* moveType = unit->moveType;
-            moveType->ProcessCollisionEvents();
+
+        process_thread_data<UnitCrushEvent>(comp.killUnits, [](UnitCrushEvent& event){
+            event.collidee->Kill(event.collider, event.crushImpulse, true);
```

Comment:
is the order here guaranteed to be deterministic? it looks like the thread walker picks the thread/vector whose first unprocessed unit with the lowest ID available, but are thread/vectors  sorted by ID for this sort of processing order to be deterministic?

Reply:
No it isn't. I've been thinking about that. The code was originally walking the unit list, which is in unit Id order, but we're walking the components instead here because not every unit is a ground walking unit. I will set the id according to the jobId, then the guarantees each thread's collection will be in ascending order.

Reply:
I will run sync tests before letting this go to mainline.

Reply:
Keep in mind this will require a synthetic test with synthetic Lua.